code/game/objects/items/weapons/RCD.dm

```diff
@@ -135,13 +135,15 @@
  * * airlock_type - an airlock typepath.
  */
 /obj/item/rcd/proc/get_airlock_image(airlock_type)
-	var/obj/machinery/door/airlock/proto = new airlock_type(null)
-	proto.icon_state = "closed"
-	if(!proto.glass)
-		proto.add_overlay("fill_closed")
-	var/icon/I = getFlatIcon(proto)
-	qdel(proto)
-	return "[icon2base64(I)]"
+	var/obj/machinery/door/airlock/airlock = airlock_type
+	var/icon/base = icon(initial(airlock.icon), "closed")
+	if(initial(airlock.glass))
+		var/icon/glass_fill = icon(initial(airlock.overlays_file), "glass_closed")
+		base.Blend(glass_fill, ICON_OVERLAY)
+	else
+		var/icon/solid_fill = icon(initial(airlock.icon), "fill_closed")
```

Comment:
maybe a dumb question, but are these able to be `get_airlock_overlay()`? LGTM otherwise

Reply:
`get_airlock_overlay` returns an image which does not work with `icon`. I could maybe use images here instead but I'm not sure its possible to build multi-sprite images.